components/identity.oauth.uma.permission.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/permission/endpoint/PermissionApiServiceImpl.java

```diff
@@ -52,18 +58,30 @@
      * @return Response with the status of the creation of a permission ticket.
      */
     @Override
-    public Response requestPermission(ResourceModelDTO requestedPermission) {
+    public Response requestPermission(ResourceModelDTO requestedPermission, MessageContext context) {
+
+        String[] tokenScopes = (String[]) ((AuthenticationContext) context.getHttpServletRequest()
+                .getAttribute(authContext)).getParameter(oauth2AllowedScopes);
+        if (!ArrayUtils.contains(tokenScopes, patScope)) {
+            log.error("Access token doesn't contain valid scope.");
+            return Response.status(Response.Status.UNAUTHORIZED).build();
+        }
 
-        PermissionService permissionService = (PermissionService) PrivilegedCarbonContext.getThreadLocalCarbonContext()
-                .getOSGiService(PermissionService.class, null);
         if (requestedPermission == null) {
             log.error("Empty request body.");
             return Response.status(Response.Status.BAD_REQUEST).build();
         }
+
```

Comment:
Better to remove extra new lines used in the code

Reply:
done